go/vt/topo/conn.go

```diff
@@ -113,6 +113,17 @@ type Conn interface {
 	// Returns ErrInterrupted if ctx is canceled.
 	Lock(ctx context.Context, dirPath, contents string) (LockDescriptor, error)
 
+	// TryLock provides exactly same functionality as 'Lock', the only difference is
+	// it tries its best to be non-blocking. With current implementation for concurrent
```

Comment:
IMO ~ `non-blocking (best-effort)` language makes no sense as something is non-blocking or it isn't. I think we should instead refer to it as only attempting to acquire the lock if it is likely to succeed. In other places we call it non-blocking w/o any caveats and that is definitely wrong. Can we just remove that wording, since IMO it does not add clarity?

If we need some short-hand way to refer to the difference between Lock and TryLock can we could instead note that the difference is that it attempts to fail-fast?

Reply:
Alternatively we could add a short timeout to the lock attempt in try lock so that it is non-blocking.

Reply:
I have changed the description as per your comments. Please review it and let me know if it is more readable now.